kernel/thread.c

```diff
@@ -47,11 +46,11 @@ void k_thread_foreach(k_thread_user_cb_t user_cb, void *user_data)
 	 * The indirect ways are through calling k_thread_create and
 	 * k_thread_abort from user_cb.
 	 */
-	key = k_spin_lock(&lock);
+	k_sched_lock();
```

Comment:
This seems like it would break SMP unfortunately if another CPU is working on _kernel.threads at the same time.

Reply:
This function is only used to read and print some debug information. It is lesser evil if we print non complete debug information than if we break bt communication during this printing.

Another way to solve it may be by the usage of the mutex, but then it has to be placed inside functions for creating and destroying the thread also.

Reply:
@andrewboie I can see from the code, that threads may not be created in ISR, but thread aborting seems to consider a situation when thread is removed from the interrupt (why?).

Any suggestion how to solve this situation without locking the whole system for the time long enough to break the bt communication?